deltacat/catalog/model/catalog.py

```diff
@@ -277,44 +255,32 @@ def put_catalog(
 
     You may explicitly provide an initialized Catalog instance, like from the Catalog constructor or
     from factory methods like Catalog.default or Catalog.iceberg
-
     Otherwise, this function initializes the catalog by using the catalog implementation provided by `impl`.
+    This function will NOT fail if the catalog already exists. It will overwrite the named catalog
 
     Args:
         name: name of catalog
         catalog: catalog instance to use, if provided
         impl: catalog module to initialize. Only used if `catalog` param not provided
-        ray_init_args: ray initialization args (used if ray must be initialized)
+        set_as_default: whether this catalog should be treated as the default.
+            This will overwrite any existing default.
+            NOTE: Catalogs with one entry will always use the single entry as the default.
+        ray_init_args: ray initialization args (used only if ray not already initialized)
     """
-    # Set namespace to default, or use value in kwargs or ray_init_args
-    if kwargs.get("namespace") is not None:
-        namespace = kwargs.get("namespace")
-        # ensure ray init arg set with this namespace
-        ray_init_args = ray_init_args if ray_init_args is not None else {}
-        ray_init_args["namespace"] = kwargs.get("namespace")
-    elif ray_init_args and "namespace" in ray_init_args:
-        namespace = ray_init_args["namespace"]
-    else:
-        namespace = _default_namespace
+    import deltacat.catalog.model.catalog as catalog_module
 
     if catalog is None:
         catalog = Catalog(impl, **kwargs)
-    elif catalog is not None and impl != DeltacatCatalog:
-        raise ValueError(
-            f"PutCatalog call provided both `impl` and `catalog` parameters. "
-            f"You may only provide one of these parameters"
-        )
 
-    if is_initialized(**{"namespace": namespace}):
-        try:
-            get_catalog(name, namespace=namespace)
-            raise ValueError(
-                f"Catalog {name} already exists in namespace '{namespace}'."
-            )
-        except KeyError:
-            # Catalog does not already exist. Add it
-            global catalog_registry
-            ray.get(catalog_registry[namespace].put.remote(name, catalog))
+    if is_initialized():
+        ray.get(catalog_module.all_catalogs.put.remote(name, catalog, set_as_default))
     else:
-        init({name: catalog}, ray_init_args=ray_init_args, **{"namespace": namespace})
+        # NOTE - since we are initializing with a single catalog, it will be set to the default
+        if not set_as_default:
+            logger.warning(
+                f"Calling put_catalog with set_as_default=False, "
+                f"but still setting Catalog {catalog} as default since it is the only catalog."
+            )
+        init({name: catalog}, ray_init_args=ray_init_args, **kwargs)
```

Comment:
Overloading `**kwargs` as input to both `init` and `Catalog` seems a little dangerous/confusing. Also, we're not actually using `**kwargs` in this case, and they could simply be omitted here.